prophy/descriptor.py

```diff
@@ -107,8 +111,7 @@ def decode_optional(parent, name, type_, data, pos, endianness, len_hints):
 
 def decode_array_delimiter(_, __, type_, data, pos, endianness, len_hints):
     value, size = type_._decode(data, pos, endianness)
-    if value < 0:
-        raise ProphyError("Array delimiter must have positive value")
+    assert value >= 0, "Array delimiter must have positive value, got %s." % value
```

Comment:
This change was done do increase coverage, but I will change it back to what it was before.

Reply:
Oh yes, I tried to cover this case with tests, but couldn't get into that `if` at all.
Negative delimiter value cannot be obtained neither by deserialization (because it causes exception) nor by assignment (raises exception as well - cannot assign negative to `prophy.u32`). As far as I know - you cannot neither change delimiter type to signed integer (`prophy.i32`).

Reply:
Then `assert` might be ok - it would fire when change in code would allow negative delimiters there.

My best attempt at reproduction
```
aurzenligl@aurzenligl-pc ~/tmp/prophy/src $ cat a.prophy 
struct X {
    i32 sizer;
    u32 x<@sizer>;
};
```
 causes:
`prophy.exception.ProphyError: X: decoded array length smaller than shift`